torch/_subclasses/fake_tensor.py

```diff
@@ -860,13 +860,24 @@ def __torch_dispatch__(self, func, types, args=(), kwargs=None):
         # and do device logic, we dont need do anything but run them
         # and ensure that Meta kernels are dispatched to (see)
         # Fake Tensor Dispatch Keys
+        if (
+            "prims::" in func._schema.name
+            and len(flat_arg_fake_tensors) != 0
+            and not hasattr(func, "prim_meta_impl")
+        ):
+            raise RuntimeError(
+                f"{func._schema.name} does not have a meta implementation"
+            )
+
         # TODO - we should be use the prim aten impl
         if (
             "prims::" in func._schema.name
             and len(flat_arg_fake_tensors) != 0
             and hasattr(func, "prim_meta_impl")
         ):
             with self:
+                # Using func(*args, **kwargs) here would cause an infinite recursion
+                # prim_meta_impl is used to avoid that
```

Comment:
Does this mean the fallback at line 902 is flawed, where we have

```
        # run kernel registered to meta for func, which include
        # python meta registrations, prims, decomps, and c++ meta fns (structured kernels)
        try:
            with in_kernel_invocation_manager(self):
                r = func(*args, **kwargs)
```

since we ended up with an infinite recursion, we won't hit the proper error message?

Reply:
Hmm removing this "if prim" and hitting that fallback actually works. But `test_nvprims_div_fake_cpu_scalar` starts to fail for `torch.ops.prims.div`.